src/tests/oAuth.test.js

```diff
@@ -0,0 +1,86 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import express from 'express';
+import dotenv from 'dotenv';
+import authController from '../controllers/authController';
+import googleFactory from '../services/factory/googleUser';
+import facebookFactory from '../services/factory/facebookUser';
+
+const {
+  expect
+} = chai;
+
+chai.use(chaiHttp);
+dotenv.config();
+
+
+const app = express();
+const router = express.Router();
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+app.use(
+  '/api/v1/auth', (req, res, next) => {
+    req.user = req.body;
+    next();
+  }, router.post('/google', authController.oAuthLogin),
+  router.post('/facebook', authController.oAuthLogin)
+);
+
+const googleUser = googleFactory.build();
+const facebookUser = facebookFactory.build();
+
+
+describe('Google oAuthentication tests', () => {
+  it('should return use is successfully created', (done) => {
+    chai
+      .request(app)
+      .post('/api/v1/auth/google')
+      .send(googleUser)
+      .end((err, res) => {
+        expect(res.body).to.be.an('object');
+        expect(res.status).to.equal(200);
+        expect(res.body.message).to.equal('User is successfully created');
+        done();
+      });
+  });
+  it('should return use is successfully logged in', (done) => {
+    chai
+      .request(app)
+      .post('/api/v1/auth/google')
+      .send(googleUser)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.message).to.equal('User is successfully logged in');
+        done();
+      });
+  });
+});
+
+
+describe('Facebook oAuthentication tests', () => {
+  it('should return user is successfully created', (done) => {
+    chai
+      .request(app)
+      .post('/api/v1/auth/facebook')
```

Comment:
it's not efficient to send unwanted requests to external APIs. Remember we have rate limiting and the way you've done it, we hit the external APIs everytime someone runs the tests